include/El/core/imports/cuda.hpp

```diff
@@ -151,7 +151,7 @@ class GPUManager
     /** Get singleton instance of CUDA manager. */
     static GPUManager* Instance();
     /** Get number of visible CUDA devices. */
-    static int NumDevices();
+    static unsigned int NumDevices();
```

Comment:
I think `unsigned int` is better than `int`, although it may be even better to use `Int` for consistency with the rest of Elemental.

Reply:
It needs to be `unsigned int` because the NVML API expects a pointer to an `unsigned int`. The compiler complains if it's just a signed int.

Reply:
I would say that, since this function returns by value and not by reference, the return type should be `El::Unsigned` to be consistent with the rest of Hydrogen (well, that typedef is declared anyway, but I don't know if it's ever actually used). The underlying data member, `numDevices_`, should of course stay as `unsigned int` to interact with the NVML interface. Unless `El::Unsigned` is an `unsigned short` or `unsigned char` (and, why would that ever happen), the conversion to `El::Unsigned` will not be a narrowing conversion.